### ISSUE TITLE
Single XLA device: Model does not train - Loss always the same

### ISSUE THREAD

**Issue by LSOFT123** [pytorch/xla#1764]
I'm not sure if it's a bug or a bad usage of API available at:
https://github.com/pytorch/xla/blob/master/API_GUIDE.md
But my EfficinetNet pre-trained model does not train with TPU enabled, same code with GPU trains correctly. With TPU loss remains the almost same after each epoch, it decreases with GPU. 

1. Install https://github.com/lukemelas/EfficientNet-PyTorch
2. Load pre-trained model (any)
3. Train it with Pytorch XLA

There is no error, it trains but loss at the end of each epoch does not move. If I display loss.item() during training I can see it moving a little bit.

```
device = xm.xla_device()
...
cnn_model= EfficientNet.from_pretrained("efficientnet-b3", num_classes=2) 
# Total params: 10,699,306
# Trainable params: 10,699,306
# Non-trainable params: 0
cnn_model.load_state_dict(torch.load(weights_path, map_location='cpu')) 
optimizer = Adam(cnn_model.parameters(), lr=lr, betas=(beta1, 0.999))
loss = nn.CrossEntropyLoss()
cnn_model= cnn_model.to(device)
cnn_model.train() # Tried to move train() before to(device) did not help
..
# Train loop for each epoch
...
for train_batch in tqdm(train_batches):
  img_data, labels_data = train_batch['image'], train_batch['label'],
  img_label = labels_data.numpy().astype(np.float)
  optimizer.zero_grad() # reset gradient
  img_data = img_data.to(device)
  labels_data = labels_data.to(device)
  input_v = Variable(img_data)
  output = cnn_model(input_v.float()) 
  loss_dis = loss(output, Variable(labels_data, requires_grad=False))
  loss_dis_data = loss_dis.item()
  loss_dis.backward() # backward pass
  xm.optimizer_step(optimizer, barrier=True)
  ...
...
# Eval model
  cnn_model.eval()
  # Compute val loss.
  ...
 # Save model if needed
 xm.save(cnn_model.state_dict(), path)
 ...
 cnn_model.train()
 ...
 # And continue training
```

Training loop returns:
[Epoch 1] Train loss: 0.6965 acc: 0.515 logloss: **0.6965** | Valid loss: 0.6833 acc: 0.563 logloss: **0.6833** | LR: 0.000200
[Epoch 2] Train loss: 0.6972 acc: 0.509 logloss: **0.6972** | Valid loss: 0.6834 acc: 0.564 logloss: **0.6834** | LR: 0.000200
[Epoch 3] Train loss: 0.6967 acc: 0.511 logloss: 0.6967 | Valid loss: 0.6832 acc: 0.565 logloss: 0.6832 | LR: 0.000200
...

I'm using nightly version as it fixes another issue about f64/f32.
I'm under Google Colab.

Python         : 3.6.9 (default, Nov  7 2019, 10:44:02) 
Numpy          : 1.17.5
PyTorch        : 1.5.0a0+bdd7dbf
PyTorchXLA     : 0.8+742abdf

**Comment by dlibenzi** [comment]
The sequence is:
1. Load model checkpoint to CPU tensors
2. Move model to XLA device
3. Create optimizer with moved model's parameters.

**Comment by LSOFT123** [comment]
It works! Thanks. I inverted steps 2 and 3. Should we move loss function to TPU too as step 4 for better performances?

`loss = loss.to(device)`

**Comment by dlibenzi** [comment]
No, do not move the loss.
Once you move the model, everything should be on XLA device.
The inputs are automatically moved by the ParallelLoader.
Also, try to limit the times you call `item()` (for example, logging loss and other metrics), once every N steps.
Like we do usually:

https://github.com/pytorch/xla/blob/7568905089fbbc16369b1a657d3a606f6f722218/test/test_train_mp_mnist.py#L129

**Comment by LSOFT123** [comment]
Yeap, noticed it for item()
